File: 0645-set-mismatch/0645-set-mismatch.py
class Solution:
    def findErrorNums(self, nums: list[int]) -> list[int]:
        n=len(nums)
        res=[0]*(n+1)
        for i in nums:
            res[i]+=1
        rep,miss=0,0
        for i in range(1,n+1):
            if res[i]==2:
                rep=i
            elif res[i]==0:
                miss=i
        return [rep,miss]

File: 0645-set-mismatch/0645-set-mismatch.java
class Solution {
    public int[] findErrorNums(int[] nums) {
        int n=nums.length;
        int freq[]=new int[n+1];
        for(int i:nums)
            freq[i]++;
            int rep=0,mis=0;
        for(int i=1;i<n+1;i++){
            if(freq[i]==2)
                rep=i;
            else if(freq[i]==0)
                mis=i;
        }
        return new int[]{rep,mis};
    }
}